src/utils/resources/vm/utils/selectors.ts

```diff
@@ -130,12 +131,20 @@ export const getAffinity = (vm: V1VirtualMachine) => vm?.spec?.template?.spec?.a
  * @param {V1VirtualMachine} vm the virtual machine
  * @returns the virtual machine boot disk
  */
-export const getBootDisk = (vm: V1VirtualMachine): V1Disk =>
-  (getDisks(vm) || [])
+export const getBootDisk = (vm: V1VirtualMachine): V1Disk => {
+  const isInstanceTypeVM = Boolean(getInstanceTypeMatcher(vm));
+
+  const rootDiskVolume = (getVolumes(vm) || []).find((volume) => volume.name === ROOTDISK);
+
+  const defaultRootDisk =
+    isInstanceTypeVM && Boolean(rootDiskVolume) ? { name: ROOTDISK } : getDisks(vm)?.[0];
+
+  return (getDisks(vm) || [])
```

Comment:
u are using getDisks(vm) 3 times, save it to a const once

Reply:
Solved